### PR TITLE
Refactor: Preparing for DEVS implementation

### PR DESCRIPTION
**Pull request by sdavtaker**
renamed functions and files related to atomic PDEVS models to start preparing for a second formalism implementation (DEVS)

### CODE REVIEW

**Comment by sdavtaker** [comment]
Fixed building support for linux in travis and required it to build.
Updated both linux and mac toolchains in Travis.

File: include/cadmium/modeling/coupling.hpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+         * Template parameter TIME is not required for coupled models, because they do not produce behavior
+         * since coupled model are structural definitions only.
+         *
+         * IP are a tuple of input_port
+         * OP are a tuple of output_port
+         * MODELS a models_tuple (the struct is a helper for simplifying the definition of tuple of templates)
+         * EICs is a tuple of EIC coupling input ports of the coupled model to input ports in submodels
+         * EOCs is a tuple of EOC coupling output ports in submodels to output ports of the coupled model
+         * ICs is a tuple IC coupling output ports in submodels to input ports in submodels
+         * SELECT is an optional template class with an imminent function that, given a list of imminent candidate
```
Comment (Laouen) [inline include/cadmium/modeling/coupling.hpp]: What is doing this SELECT docstring in the pdevs coupled_model structure?

**Review by Laouen — COMMENTED** [review]
I didn't see anything to change in the code, I like the namespace renames and the coverage is now much better than before.

I would use a standard docstring format to automate the documentation.  A good doctstring format is Doxigen. Currently I think you are jsut commenting structs without a specific format.

If you won't follow the doxigen suggestion just tell me and I merge the PR.

File: include/cadmium/modeling/coupling.hpp (hunk 1 above)
Comment (sdavtaker) [reply include/cadmium/modeling/coupling.hpp]: Fixed, those come from an error in a commit. Solved. 
About the doc format, I think we should discuss that one offline and when deciding which way to go fix it in a different PR.

**Pull request merged**